### PR TITLE
Adding the igrafx mining sdk 

### PR DESCRIPTION
**Pull request by Dourouve**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Dourouve** [comment]
@conda-forge/help-python All checks have passed. The recipe is ready for your review

**Comment by Dourouve** [comment]
@conda-forge/help-python
It has been 3 months since this package passed all its tests and it does not yet have a review.

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    - requests ==2.29.0
+    - pydruid ==0.6.5
+    - pandas ==1.3.5
+    - SQLAlchemy ==2.0.12
+    - networkx ==2.6.3
+    - numpy ==1.21.6
+    - toml ==0.10.2
+    - python-dotenv ==0.19.2
+    - pytest ==7.3.1
+    - pytest-dependency ==0.5.1
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:40]: Can you please update these dependencies according to https://github.com/igrafx/mining-python-sdk/blob/7ef88c2e9f11bcfe1334d03ddcefbffb12e5a639/pyproject.toml#L12 and not make them as strict as you currently have it?

**Review by BastianZim — CHANGES_REQUESTED** [review]

**Comment by Dourouve** [comment]
@conda-forge/help-python @BastianZim  All checks have passed and the requested changes have been made. The recipe is ready for your review. Thanks.

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  build:
+    - python >=3.10.0,<3.12.1
+    - pip
+    - poetry ==1.7.1
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:22]: As your package is noarch, this section is not needed.

```suggestion
```

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  host:
+    - pip
+    - python >=3.10.0,<3.12.1
+    - poetry ==1.7.1
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:26]:
```suggestion
    - poetry-core
```

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    - SQLAlchemy >=2.0.29
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:27]: Is this really needed here?

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  run:
+    - python >=3.10.0,<3.12.1
+    - poetry ==1.7.1
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:30]: I don't see this is the requirements.

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    - requests >=2.30.0
+    - pydruid >=0.6.5
+    - pandas >=2.0.0,<=2.2.1
+    - SQLAlchemy >=2.0.29
+    - networkx >=3.0.0,<3.3.0
+    - numpy >=1.25.0
+    - toml ==0.10.2
+    - python-dotenv ==1.0.1
+    - pytest ==8.1.1
+    - pytest-dependency >=0.5.1
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:40]: I don't think these are required for running. If they are just needed for the test, specifying there is sufficient.

File: recipes/igrafx-mining-sdk/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+test:
+  source_files:
+    - igrafx_mining_sdk/
+    - tests/
+  imports:
+    - igrafx_mining_sdk
+    - pytest
+  commands:
+    - pytest ./tests/test_01_column_mapping.py
```
Comment (BastianZim) [inline recipes/igrafx-mining-sdk/meta.yaml:50]:
```suggestion
    - pytest ./tests/test_01_column_mapping.py
    - pip check
```
To check dependencies

**Review by BastianZim — CHANGES_REQUESTED** [review]
Just a couple more things.

**Comment by Dourouve** [comment]
@conda-forge/help-python @BastianZim I have modified the recipe according to your suggestions. All checks are passing. Thanks.

**Comment by Dourouve** [comment]
@conda-forge/help-python @BastianZim 
It has been a while months since this package passed. Please review.

**Comment by xhochy** [comment]
Please address the linter comments

**Review by xhochy — APPROVED** [review]

**Pull request merged**